### DISCUSSION TITLE
Custom fields / columns

### DISCUSSION THREAD

**Discussion by KodeXall** [keepassxreboot/keepassxc#9299] [Ideas]
I propose adding an option of creating of custom fields applied to all entries, which then would be seen and sorted out in as additional columns

Reasoning behind this is that some services allow for using of the e-mail address as the login - while others do not. And so if a user has multiple e-mail accounts and dozens of credentials for various on-line services in the database, then in order to keep an easy [visual] track of which e-mail account user has used how many times so far, all such user has to do is to sort the column of **E-mail** - which unfortunately does not exist. The same is with e.g. **Telephone Number**: how is a user suppose keep track which number was given to which service, if any? Some services just allow for adding telephone credentials while others plain require. And what about keeping track of e-mail addressees added o services as recovery ones. And so on and so on. So if a user changes or looses one of the e-mails accounts or telephone numbers, then going through all of multiple entries would be much easier if a user could see a sorted list of a given extra data

And yes: there is the **Search** box - but you cannot see all of the data or even sort out column of **Notes** if our entries in this field have more than just 1 line od data. So right now user first narrows down lists of entries and then is forced to enter a credential entry in order to read just a part of **Notes**

Thus a need for adding custom fields and columns, which would allow for dividing important data among separate fields

**Comment by KodeXall** [comment]
> Thus a need for adding custom fields and columns, which would allow for dividing important data among separate fields

A user can utilize in any **Entry** its **Advanced** tab which could seem like a workaround. But then again: such **Additional Attributes** cannot be shared between **Entries** and they do not show up as new columns with possibility to turn them on and off - and thus it is impossible to conveniently see later on made-up by user fields and sort them by theirs data

**Reply by arran4** [reply]
Being able to have an email as well as a username would be great as I generate email addresses.

**Reply by ediw8311xht** [reply]
I agree. Online accounts may have a username and an email, and may require a specific one to login. For me personally, this has made have a consistent database harder since I will use the "Username" field to contain the email/username I used for signing in, and then I have to manually add an additional field for the other one.

**Reply by AnonBlankzy** [reply]
I usually just put them in the Notes section

**Comment by novasbc** [comment]
I agree with this feature request!

Many sites do not use email for login, so I'm forced to put it in the notes field, but almost all sites require a valid email address.

> Being able to have an email as well as a username would be great as I generate email addresses.

I do the same, and it's more common than you think!

There's an [existing feature request ](https://github.com/keepassxreboot/keepassxc/issues/1493)for this as well.

**Comment by Quayron** [comment]
well i totally agree it needs to have feature so we can add custom colums, for ease of use so it can be applied to other fields 
like in my case 
i wanna add others informations comes with accounts that also helps you to recover in future problems 
like 
-lastname
-date of birth
-email
-username and accound id two different thigs i would like to add
-prone number

but it has to be open to custom because i dont know what else i wanna add later on
